src/en/administration/from_scratch.md

```diff
@@ -0,0 +1,886 @@
+
+```
+
+Install extra packages required for pict-rs:
+
+| packages    |                                                  |
+|-------------|--------------------------------------------------|
+| media-libs  | [gexiv2](https://gitlab.gnome.org/GNOME/gexiv2)  |
+| media-gfx   | [imagemagick](https://imagemagick.org/index.php) |
+| media-video | [ffmpeg](https://ffmpeg.org/)                    |
+| sys-devel   | [clang](https://clang.llvm.org/)                 |
+
+Packages required for pict-rs (in case on separate system):
+
+| packages |                                    |
+|----------|------------------------------------|
+| dev-lang | [rust](https://www.rust-lang.org/) |
+
+Script this or it has to be run manually as user...?
+
+```bash
+su - pictrs
+git clone https://git.asonix.dog/asonix/pict-rs.git
+cd pict-rs
+git tag -l
+git checkout tags/v0.2.5-r0
+
+cargo build --release
+cp target/release/pict-rs .
+cd ~pictrs
+mkdir pictrs-data
+```
+
+**Something missing in the pict-rs README - it created & uses a pict-rs folder in /tmp**
+
+If you do anything funky like I have (changing the user pict-rs runs under) and you wind up with permission problems (which the logs don't tell you *what* it's having a permission issue with), this could be your issue. Also, time will tell whether this folder gets cleaned up appropriately or not.
+
+Running pictrs is along the lines of:
+
+```bash
+pict-rs/pict-rs -a 127.0.0.1:9000 -p ~pictrs/pictrs-data/
+```
+
+But we'll just use the [init script](#initpict-rs). At this point, fire everything up via the [init scripts](#init-scripts), should all go. Setup the init scripts to run at boot time. Presumably you've [setup nginx as per examples](#nginx-script) and you can reach your instance.
+
+---
+
+## Updating
+
+```bash
+su - lemmy
+cd lemmy
+```
+
+BACKUP [config/config.hjson](#config-script) somewhere
+
+```bash
+git fetch
+git tag -l
+git checkout tags/WHATEVER
+
+cargo build --release
+
+cd ~/lemmy-ui
+```
+
+#### List the tags/releases available
+
+```bash
+git fetch
+git tag -l
+git checkout tags/WHATEVER
+git submodule update --remote
+```
+
+### Build the frontend
+
+#### This is for prod
+
+```bash
+yarn install --pure-lockfile   # Is this step really needed?
+#yarn upgrade --pure-lockfile  # ?? Did I get it?
+#yarn   # Is this step really needed?  One of these steps is for sure. (Should be unnecessary)
+yarn build:prod
+```
+
+restart lemmy-ui
+
+### iFramely update
+
+```bash
+su - iframely
+cd iframely
+git fetch
+git tag -l
+git checkout tags/v1.6.0  # Or whatever the current appropriate release is
+npm install
+```
+
+restart iframely
+
+### pict-rs update
+
+```bash
+su - pictrs
+cd pict-rs
+git fetch
+git tag -l
+git checkout tags/v0.2.5-r0  # (or whatever is currently mentioned in the lemmy docker file)
+
+cargo build --release
+cp target/release/pict-rs .
+```
+
+restart pictrs
+
+---
+
+## Index: scripts
+
+### config script
```

Comment:
This is the configuration, not a script. Same with iframely below. Only the init scripts are actually scripts (cause they contain bash code).